src/Middleware/Diagnostics/src/DeveloperExceptionPage/DeveloperExceptionPageMiddleware.cs

```diff
@@ -127,13 +128,22 @@ public async Task Invoke(HttpContext context)
         // Assumes the response headers have not been sent.  If they have, still attempt to write to the body.
         private Task DisplayException(ErrorContext errorContext)
         {
-            var compilationException = errorContext.Exception as ICompilationException;
-            if (compilationException != null)
+            var httpContext = errorContext.HttpContext;
+            var acceptHeader = httpContext.Request.Headers[HeaderNames.Accept].ToString();
+
+            // If the client does not ask for HTML just format the exception as plain text
+            if (acceptHeader != null && !acceptHeader.Contains("text/html", StringComparison.OrdinalIgnoreCase))
+            {
+                httpContext.Response.ContentType = "text/plain";
+                return httpContext.Response.WriteAsync(errorContext.Exception.ToString());
```

Comment:
What happened to the idea of using the same model and rendering most of the same content but as text?

Reply:
I didn't see the benefit.

Reply:
Of including the headers, cookies, etc? They're just as relevant to the API scenario.

Reply:
So you want to dump all of that information verbatim split by new lines or something? I don't have a problem with that but maybe instead of dumping all of that information, we can decide what is relevant (and doesn't come back in the HTTP response itself).

Reply:
Right. We do something similar to debug reverse proxies, IIS, etc..
https://docs.microsoft.com/en-us/aspnet/core/host-and-deploy/proxy-load-balancer?view=aspnetcore-2.2#troubleshoot
https://github.com/aspnet/AspNetCore/blob/c6df1b8a14858ede60ccb530022f9201511a1c40/src/Servers/IIS/IIS/samples/NativeIISSample/Startup.cs#L32-L101

Note the request headers are relevant because they can be transformed by intermediaries and middleware. You want to know what the server saw.

Reply:
Updated it to add request headers